src/displayBackend/common/ConnectorBase.cpp

```diff
@@ -19,12 +19,329 @@
+
+	/* 16-bit value made up of three 5-bit letters, big-endian. */
+	uint16_t vendorId = (((('X' - '@') & 0x1f) << 10) |
+						 ((('E' - '@') & 0x1f) <<  5) |
+						 ((('N' - '@') & 0x1f) <<  0));
+#ifdef IS_BIG_ENDIAN
+	edidBlock->mfg_id[0] = vendorId & 0xff;
+	edidBlock->mfg_id[1] = vendorId >> 8;
+#else
+	edidBlock->mfg_id[1] = vendorId >> 8;
+	edidBlock->mfg_id[0] = vendorId & 0xff;
+#endif
+
+	/* Manufacturer product code. 16-bit number, little-endian. */
+#ifdef IS_BIG_ENDIAN
+	edidBlock->prod_code[0] = 'M';
+	edidBlock->prod_code[1] = 'V';
+#else
+	edidBlock->prod_code[0] = 'M';
+	edidBlock->prod_code[1] = 'V';
+#endif
+
+	/* Serial number. 32 bits, little-endian. */
+	edidBlock->serial = toLittleEndian32(0x20200505);
+
+	/* Week of manufacture - not consistent between manufacturers. */
+	edidBlock->mfg_week = 1;
+	/* Year of manufacture. Year = datavalue + 1990. */
+	edidBlock->mfg_year = 2020 - 1990;
+
+	/* EDID version and revision: advertise 1.4. */
+	edidBlock->version = 1;
+	edidBlock->revision = 4;
+
+	/*
+	 * Video input parameters bitmap: digital, 8 bits per color,
+	 * display port.
+	 */
+	edidBlock->input = DRM_EDID_INPUT_DIGITAL | DRM_EDID_DIGITAL_DEPTH_8 |
+		DRM_EDID_DIGITAL_TYPE_DP;
+
+	/* Set width and hight as undefined. */
+	edidBlock->width_cm = 0;
+	edidBlock->height_cm = 0;
+
+	/* Display gamma, 2.2. */
+	edidBlock->gamma = 220 - 100;
+
+	/* Supported features bitmap. */
+	edidBlock->features = DRM_EDID_FEATURE_PREFERRED_TIMING |
+		DRM_EDID_FEATURE_STANDARD_COLOR;
+}
+
+static uint32_t edid_to_10bit(float value)
+{
+	return (uint32_t)(value * 1024 + 0.5);
+}
+
+void ConnectorBase::edidPutColorSpace(edid* edidBlock)
+{
+	const float rx = 0.6400;
+	const float ry = 0.3300;
+	const float gx = 0.3000;
+	const float gy = 0.6000;
+	const float bx = 0.1500;
+	const float by = 0.0600;
+	const float wx = 0.3127;
+	const float wy = 0.3290;
+
+	uint32_t red_x   = edid_to_10bit(rx);
+	uint32_t red_y   = edid_to_10bit(ry);
+	uint32_t green_x = edid_to_10bit(gx);
+	uint32_t green_y = edid_to_10bit(gy);
+	uint32_t blue_x  = edid_to_10bit(bx);
+	uint32_t blue_y  = edid_to_10bit(by);
+	uint32_t white_x = edid_to_10bit(wx);
+	uint32_t white_y = edid_to_10bit(wy);
+
+	edidBlock->red_green_lo = (((red_x   & 0x03) << 6) |
+							   ((red_y   & 0x03) << 4) |
+							   ((green_x & 0x03) << 2) |
+							   ((green_y & 0x03) << 0));
+
+	edidBlock->black_white_lo = (((blue_x  & 0x03) << 6) |
+								 ((blue_y  & 0x03) << 4) |
+								 ((white_x & 0x03) << 2) |
+								 ((white_y & 0x03) << 0));
+
+	edidBlock->red_x = red_x >> 2;
+	edidBlock->red_y = red_y >> 2;
+
+	edidBlock->green_x = green_x >> 2;
+	edidBlock->green_y = green_y >> 2;
+
+	edidBlock->blue_x = blue_x >> 2;
+	edidBlock->blue_y = blue_y >> 2;
+
+	edidBlock->white_x = white_x >> 2;
+	edidBlock->white_y = white_y >> 2;
+}
+
+void ConnectorBase::edidPutStandardTiming(edid* edidBlock, int index,
+										  uint32_t xres, uint32_t yres)
+{
```

Comment:
If you have detailed timings you can completely drop standard timings part. I'd even recommend dropping this, it is misleading

Reply:
will drop

Reply:
dropped